### ISSUE TITLE
Deployment to cloud foundry fails

### ISSUE THREAD

**Issue by drissamri** [jhipster/generator-jhipster#2225]
After running jhipster:cloudfoundry (Bluemix) on version 2.23 this error occurs:

```
Uploading ucll...
Done uploading
FAILED
Error uploading application.
The app package is invalid: Could not zip the package
 STDOUT: "
zip error: Nothing to do! (try: zip -q -r --symlinks /var/vcap/data/cloud_controller_ng/tmp/safezipper20151026-11093-1r8q6j1/package.zip . -i .)
"
 STDERR: ""
```

A blank Spring Boot jar application works fine, I can't make out why a JHipster war is failing.

**Comment by drissamri** [comment]
The same error is logged at https://github.com/cloudfoundry/cli/issues/634 but this is about when using symlinks. I don't use symlinks and I don't think JHipster makes any symlinks automatically?

**Comment by drissamri** [comment]
I have just tried this with version generator version 2.21.1 and it works, it doesn't seem to work with newer versions.

**Comment by jdubois** [comment]
It might be related to Spring Boot 1.3.0.RC1, that's the biggest change we did

**Comment by pkcool** [comment]
there same happening here. couldn't deploy to cloud foundry (Pivotal web services) with version 2.23. I had to roll back to version 2.22 which works well.

**Comment by jdubois** [comment]
Oh looks like this issue: https://github.com/cloudfoundry/cli/issues/411

-> I'm going to work on this for the next hour, I'll post feedback here

**Comment by jdubois** [comment]
OK this is a bug in Spring Boot 1.3 as written in my previous comment.

I have a workaround, using `<executable>false</executable>` in the Maven Spring Boot plugin. This has a number of consequences (you can't launch the WAR as easily as before), but this isn't a big issue

-> I'm going to change this in our code, and modify the documentation accordingly

**Comment by philwebb** [comment]
@jdubois FWIW the CF bug is the primary reason that we didn't enable fully executable jars by default.

**Comment by jdubois** [comment]
Thanks @philwebb -> I expect to do a new release tomorrow that will fix the issue for us

**Comment by drissamri** [comment]
Thank you @jdubois, this solves the issue indeed.

**Comment by cbornet** [comment]
Can someone test if executable=true works with the latest CF CLI ( v6.14.1) ?

**Comment by ariscapellos** [comment]
FWIW - This spring boot 'true' prod setting has been working fine for me with CF CLI ( 6.15.0+fa1bfe2-2016-01-13 ) and jhipster 2.27.0  

Thanks for this sterling generator!

**Comment by ajinkyaghorpade** [comment]
Still getting same issue with jhipster 2.27.2

**Comment by jdubois** [comment]
This is an old issue, it is solved, so you are probably not having the "same" issue. Please open up a new ticket with the details.
